### DISCUSSION TITLE
How to set the filter multi-parameters of http request, I report error 400

### DISCUSSION THREAD

**Discussion by dong-lufei** [pocketbase/pocketbase#3172] [Q&A]
```
await http.get("/api/collections/settings/records?filter=(a='x')", {...})
await http.get("/api/collections/settings/records?filter=(b='y')", {...})
```
The settings of the above two requests return correct data, but I return 400 with this combination, and report an invalid filter parameter
```
await http.get("/api/collections/settings/records?filter=(a='x'&&b='y')", {...})
```

**Comment by ganigeorgiev** [comment]
The query parameters need to be properly url encoded, aka. try with:

```js
await http.get("/api/collections/settings/records?filter=a%3D%27x%27%20%26%26%20b%3D%27y%27", {...})
```

**Comment by dong-lufei** [comment]
@ganigeorgiev 
```
const filterParas = encodeURIComponent(`a='x'&&b='y'`)
await http.get(`/api/collections/settings/records?filter=(${filterParas })`, {...})
```
Use encodeURIComponent encoding to solve, but filter only one condition why not convert the encoding can also be, and multiple conditions but to convert the encoding (the document did not write to turn)

**Reply by ganigeorgiev** [reply]
I'm not sure I follow.

The single filter expression may work because it doesn't have `&&` or other conflicting characters, but this is not guarantee to always be the case so **if you are not using the SDKs you must always url encode the query parameters** (_including the parenthesis in your example, if you really need them_).
